modules/mongodb/collect.go

```diff
@@ -75,3 +75,32 @@ func (m *Mongo) metricExists(iface interface{}, chart *module.Chart) {
 		return
 	}
 }
+
+// dbStatsCollect creates the map[string]int64 for the available dims
+// it calls listDatabase and then dbstats for each database internally
+func (m *Mongo) dbStatsCollect() map[string]int64 {
+	if m.databasesMatcher == nil {
+		return map[string]int64{}
+	}
+	databases, err := m.mongoCollector.listDatabaseNames()
+	if err != nil {
+		return map[string]int64{}
+	}
+
+	// add dims for each database
+	m.dimsForDbStats(databases)
+	m.databases = databases
+
+	ms := map[string]int64{}
+	for _, database := range databases {
+		if !m.databasesMatcher.MatchString(database) {
+			continue
+		}
+		stats, err := m.mongoCollector.dbStats(database)
+		if err != nil {
+			return map[string]int64{}
```

Comment:
```suggestion
		stats, err := m.mongoCollector.dbStats(database)
		if err != nil {
			return nil
```

And we need to log the error